pkg/interop/native/ledger/block.go

```diff
@@ -29,3 +29,27 @@ type Block struct {
 	// TransactionsLength represents the length of block's transactions array.
 	TransactionsLength int
 }
+
+// BlockSR is a stateroot-enabled Neo block. It's returned from the Ledger contract's
+// GetBlock method when StateRootInHeader NeoGo extension  is used. Use it only when
+// you have it enabled when you need to access PrevStateRoot field, Block is sufficient
+// otherwise. To get this data type ToBlockSR method of Block should be used. All of
+// the fields are same as in Block except PrevStateRoot.
+type BlockSR struct {
```

Comment:
Maybe we also need to adjust `ledger.GetBlock` signature so that it returns `interface{}` that can be converted either to `*Block` or to `*BlockSR`?

Reply:
The idea is to avoid that (see #2551 also). 99% of the time we don't care about `BlockSR`, so forcing users to cast return values doesn't seem right. When we do, we use the conversion function.